src/blockchainApiConnection/__mocks__/BlockchainApiConnection.ts

```diff
@@ -5,12 +5,264 @@
+ * This result defaults to `Finalized`, a default which can be changed by means of the `__setDefaultResult()` function:
+ * ```
+ * require('../blockchainApiConnection/BlockchainApiConnection').__setDefaultResult(
+ *   false
+ * )
+ * const transfer = blockchain.api.tx.balances.transfer(alice.address, amount) // returns a mock SubmittableExtrinsic that has a send() method
+ * const result = await blockchain.submitTx(alice, transfer)                   // calls transfer.send() internally
+ * ```
+ * You can also queue results with
+ * ```
+ * require('../blockchainApiConnection/BlockchainApiConnection').__queueResults(
+ *   [true, false]
+ * )
+ * ```
+ * After queue results have been consumed via tx calls, the mock implementation will resume returning the default result.
+ *
+ * Mocked query methods return representations of 'not present' by default, such as Option(..., null) or Vec(..., []).
+ * You can set different return values during test execution by importing __mocked_api, then calling jest's return
+ * value setters `.mockReturnValue` or `.mockReturnValueOnce` on the method you want to modify:
+ * ```
+ *   const mocked_api = require('../blockchainApiConnection/BlockchainApiConnection').__mocked_api
+ *   mocked_api.query.delegation.children.mockReturnValue(
+ *     new Vec(
+ *       H256,
+ *       ['0x123', '0x456', '0x789']
+ *     )
+ *   )
+ * ```
+ * `.mockReturnValue` changes the default return value for the scope of the current test file.
+ * `.mockReturnValueOnce()` can be called multiple times to build a queue of return values. After the queue has
+ * been emptied by calling the query in question repeatedly, jest will return the default value again.
+ *
+ */
+
+import Blockchain, { IBlockchainApi } from '../../blockchain/Blockchain'
+import { ApiPromise, SubmittableResult } from '@polkadot/api'
+import { Option, Tuple, Vec, H256, u64, u128 } from '@polkadot/types'
+import { SubmittableExtrinsic } from '@polkadot/api/promise/types'
+import { ExtrinsicStatus } from '@polkadot/types/interfaces'
+import AccountId from '@polkadot/types/primitive/Generic/AccountId'
+
+const BlockchainApiConnection = jest.requireActual('../BlockchainApiConnection')
 
-export async function getCached() {
-  return Promise.resolve(Blockchain)
+async function getCached(
+  _: string = BlockchainApiConnection.DEFAULT_WS_ADDRESS
+): Promise<IBlockchainApi> {
+  if (!BlockchainApiConnection.instance) {
+    BlockchainApiConnection.instance = Promise.resolve(
+      new Blockchain(__mocked_api as ApiPromise)
+    )
+  }
+  return BlockchainApiConnection.instance
 }
 
-export default getCached
+const TxResultsQueue: SubmittableResult[] = []
+let defaultTxResult: SubmittableResult = __makeSubmittableResult(true)
+
+class MockSubmittableExtrinsic {
+  result: SubmittableResult
+  method: string = 'mock tx'
+
+  constructor(result: SubmittableResult) {
+    this.result = result
+  }
+
+  public sign(_a: any, _b: any, _c: any) {
```

Comment:
The input variables could be removed as they are not used at all.

Reply:
These are mock implementations, I thought they have to have a signature identical to the original? Won't it just break otherwise?

Reply:
you're right, that works. Have updated!